app/views/incomeSources/add/IncomeSourceCheckDetails.scala.html

```
@@ -0,0 +1,208 @@
+    }
+
+    @changeAccountingMethodURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethodAgent(viewModel.incomeSourceType).url
+        } else {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethod(viewModel.incomeSourceType).url
+        }
+    }
+
+@mainTemplate(
+    pageTitle = getMessage("title"),
+    backUrl = Some(backUrl),
+    useFallbackBackLink = true,
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @hmrcPageHeading(PageHeading(
+        text = getMessage("title"),
+        headingClasses = Some("govuk-heading-l"),
+        section = Some(getMessage("heading")),
+        captionClasses = Some("govuk-caption-l")
+    ))
+
+    @formWithCSRF(action = postAction) {
+
+        <dl class="govuk-summary-list" id="check-details-table">
+            @if(viewModel.incomeSourceType == SelfEmployment){
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-name")</dt>
+                    <dd class="govuk-summary-list__value" id="business-name-value">@{
+                        viewModel.businessName.getOrElse("")
+                    }</dd>
+                    <dd class="govuk-summary-list__actions govuk-!-text-align-right govuk-!-margin-0">
+                        <a class="govuk-link" href=@changeBusinessNameURL>
+                        @getMessage("change")
+                        </a>
+                    </dd>
+                </div>
+            }
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">@getMessage("start-date")</dt>
+                <dd class="govuk-summary-list__value" id="start-date-value">@{
+                    viewModel.businessStartDate.get.toLongDate
```

if we can avoid calling .get and .head in the view? do we have a default value or we should raise an exception when constructing model ? it seem that we this field should not be Optional;

if this is different across view (entities: in some Optional and in some mandatory ) we would have to use different model definition for this